nav2_behavior_tree/plugins/decorator/is_path_longer_on_approach.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2022 Neobotix GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+#include <vector>
+#include "nav2_util/geometry_utils.hpp"
+
+#include "nav2_behavior_tree/plugins/decorator/is_path_longer_on_approach.hpp"
+
+namespace nav2_behavior_tree
+{
+
+IsPathLongerOnApproach::IsPathLongerOnApproach(
+  const std::string & name,
+  const BT::NodeConfiguration & conf)
+: BT::DecoratorNode(name, conf)
+{
+  node_ = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
+}
+
+bool IsPathLongerOnApproach::isPathUpdated(
+  nav_msgs::msg::Path & new_path,
+  nav_msgs::msg::Path & old_path)
+{
+  return new_path != old_path && old_path.poses.size() != 0 &&
+         new_path.poses.size() != 0 &&
+         old_path.poses.back() == new_path.poses.back();
+}
+
+bool IsPathLongerOnApproach::isRobotInGoalProximity(
+  nav_msgs::msg::Path & old_path,
+  double & prox_leng)
+{
+  return nav2_util::geometry_utils::calculate_path_length(old_path, 0) < prox_leng;
+}
+
+bool IsPathLongerOnApproach::isNewPathLonger(
+  nav_msgs::msg::Path & new_path,
+  nav_msgs::msg::Path & old_path,
+  double & length_factor)
+{
+  return nav2_util::geometry_utils::calculate_path_length(new_path, 0) >
+         length_factor * nav2_util::geometry_utils::calculate_path_length(
+    old_path, 0);
+}
+
+inline BT::NodeStatus IsPathLongerOnApproach::tick()
+{
+  getInput("new_path", new_path_);
+  getInput("old_path", old_path_);
+  getInput("prox_leng", prox_leng_);
+  getInput("length_factor", length_factor_);
+
+  if (old_path_.poses.size() != 0) {
+    first_time_ = false;
+  }
+
+  // Check if the path is updated and valid, compare the old and the new path length,
+  // given the goal proximity and check if the new path is longer
+  if (isPathUpdated(new_path_, old_path_) && isRobotInGoalProximity(old_path_, prox_leng_) &&
+    isNewPathLonger(new_path_, old_path_, length_factor_) && !first_time_)
+  {
+    const BT::NodeStatus child_state = child_node_->executeTick();
+    switch (child_state) {
+      case BT::NodeStatus::RUNNING:
+        return BT::NodeStatus::RUNNING;
+      case BT::NodeStatus::SUCCESS:
+        first_time_ = true;
```

Comment:
Wouldn't this be trivially overridden as `false` in L67 because the old path is still populated? I don't think this line has any effect and can be removed

Reply:
Just to give you a clarity, this has been already taken care and it is not relevant anymore.